.github/workflows/haskell.yml

```diff
@@ -465,3 +465,27 @@ jobs:
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
           SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+
+  SRP-validity:
+    name: Check if formal-ledger-specifications SRP is valid
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Check if provided formal-ledger-specifications SRP commit hash exists
+      run: |
+        TAG=$(sed -ne \
+          '/ *location: *https:\/\/github.com\/IntersectMBO\/formal-ledger-specifications.git/,/--sha256/ p' \
+          cabal.project \
+        | sed -ne 's/^ *tag: *//p')
+        # The unshallow option is required so we fetch
+        # the full history of the `MAlonzo-code` branch.
+        git fetch --unshallow https://github.com/intersectmbo/formal-ledger-specifications.git MAlonzo-code
```

Comment:
Maybe you already tested this, but I feel like the `--unshallow` isn't necessary because the fetch is coming from a separate repository that has no commits in common with this one, and so no commits will be suppressed.

It doesn't do any harm to leave it in there, of course, but if it's not needed then it does make things look more complicated than they need to be.

Reply:
Interesting! Didn't know that. Indeed, it isn't required here, I've just checked it. Cheers!